kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusMetrics.java

```diff
@@ -87,4 +89,24 @@ Counter getDMNNumberOfEvaluationFailed() {
     Histogram getDroolsEvaluationTimeHistogram() {
         return droolsEvaluationTimeHistogram;
     }
+
+
+    private static final Summary optaPlannerSolverDuration = Summary.build()
```

Comment:
@vblagoje how about defining buckets, don't you need to add them beforehand?

Reply:
@cristianonicolai Yes, I would love to define buckets but it's hard as we don't know how long will solvers run. It could be anywhere from seconds to hours. If we had some data about the problem a customer is trying to solve and some beforehand knowledge about their duration then yes histograms would make sense. But in this case, I think summaries make more sense and then we just pick the standard quantiles/percentiles so whatever solver durations are - we are ok. I am still open to fine-tuning this idea and to see how actual metrics data comes out so we can pick the right metric type.

Reply:
I picked summary after I read this [page](https://prometheus.io/docs/practices/histograms/) perhaps more than 10 times to understand the difference between histograms and summaries :-) LMK if you think otherwise.

Reply:
@vblagoje I feel your pain, I also read that page many times and I had the same problems with the buckets defined in Drools. 
One way could be let the user specify the buckets in the configuration but I haven't explored that possibility yet.

Reply:
@vblagoje @lucamolteni exactly the same problem I'm facing, that's why I was curious about your approach. Good to know that we're all on the same boat :)